c7n/resources/iam.py

```diff
@@ -41,6 +42,80 @@ class User(QueryResourceManager):
     resource_type = 'aws.iam.user'
 
 
+@resources.register('iam-policy')
+class Policy(QueryResourceManager):
+
+    resource_type = 'aws.iam.policy'
+
+
+@resources.register('iam-profile')
+class InstanceProfile(QueryResourceManager):
+
+    resource_type = 'aws.iam.instance-profile'
+
+
+@resources.register('iam-certificate')
+class ServerCertificate(QueryResourceManager):
+
+    resource_type = 'aws.iam.server-certificate'
+
+
+@InstanceProfile.filter_registry.register('stale')
+class StaleInstanceProfiles(ValueFilter):
+
+    schema = type_schema('stale', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+
+        def _get_last_event(resource):
+            client = local_session(
+                self.manager.session_factory).client('cloudtrail')
+            events = client.lookup_events(
```

Comment:
this isn't viable. lookup events has many, many caveats, rate limits (1 per second per account), CUD only apis (no describe read calls), 7 day history. its not appropriate for this use case. instead we need to look at any resource that can have iam role associated, same we do security group unused filter for example.